nemo/collections/nlp/models/language_modeling/lm_model.py

```diff
@@ -51,34 +53,42 @@ def output_types(self) -> Optional[Dict[str, NeuralType]]:
             'nsp_logits': self.nsp_classifier.output_types['logits'],
         }
 
-    def __init__(
-        self,
-        config_file: Optional[str] = None,
-        pretrained_model_name: Optional[str] = 'bert-base-uncased',
-        preprocessing_args: Optional[Dict[str, str]] = None,
-    ):
-        """
-        Args:
-            config_file: model config file
-            pretrained_model_name: BERT model name 
-            preprocessing_args: preprocessing parameters for on the fly data preprocessing
-        """
-        # TODO: This is a workaround - please fix - see text_classification_model or asr for example
-        cfg = DictConfig(
-            {
-                'config_file': config_file,
-                'pretrained_model_name': pretrained_model_name,
-                'preprocessing_args': preprocessing_args,
-            }
-        )
-        super().__init__(cfg=cfg)
-        self.pretrained_model_name = pretrained_model_name
-        self.tokenizer = None
-        if preprocessing_args:
-            self.__setup_tokenizer(preprocessing_args)
-        self.bert_model = get_pretrained_lm_model(pretrained_model_name=pretrained_model_name, config_file=config_file)
+    def __init__(self, cfg: DictConfig, trainer: Trainer = None):
+        if cfg.language_model.bert_config_file is not None:
+            logging.info(
+                (
+                    f"HuggingFace BERT config file found. "
+                    f"LM will be instantiated from: {cfg.language_model.bert_config_file}"
+                )
+            )
+            self.vocab_size = json.load(open(cfg.language_model.bert_config_file))['vocab_size']
+        elif cfg.language_model.bert_config.vocab_size is not None:
+            self.vocab_size = cfg.language_model.bert_config.vocab_size
+        else:
+            self.vocab_size = None
```

Comment:
vocab size is set here, then model is loaded and then vocab size is extracted from model at the end. I think it is better to first get_model, then use the vocab_size of the model to create the tokenizer. The whole config loading is duplicated here.

Reply:
Yup, I like that better.

Reply:
Unfortunately, I don't think we can do it that way, since we need call super after instantiating the tokenizer but before instantiating the bert model.